src/jit/lsra.cpp

```diff
@@ -6836,7 +6919,20 @@ void LinearScan::freeRegister(RegRecord* physRegRecord)
             // we wouldn't unnecessarily link separate live ranges to the same register.
             if (nextRefPosition == nullptr || RefTypeIsDef(nextRefPosition->refType))
             {
+#ifdef _TARGET_ARM_
+                if (assignedInterval->registerType == TYP_DOUBLE && !genIsValidDoubleReg(physRegRecord->regNum))
+                {
+                    // Don't call unassignPhysReg() again, because two FP registers of a double reister are already
+                    // unassigned together when this function is called with valid double register number.
+                    assert(assignedInterval->assignedReg != physRegRecord);
+                }
+                else
+                {
+                    unassignPhysReg(physRegRecord, nullptr);
+                }
+#else
```

Comment:
In my view, it would be better to just put the "then" clause under the `#ifdef _TARGET_ARM_` (through the `else' line). This makes it clearer that the non-double case is common with the handling for other targets.